datasets.py

```diff
@@ -36,122 +40,137 @@ class AudioTextDataset(torch.utils.data.Dataset):
-
-			if duration_ok and exclude_ok:
-				b = bucket(example) if bucket is not None else 0
-				for t in example:
-					t['bucket'] = b
-					t['ref'] = t.get('ref', transcripts.ref_missing)
-					t['begin'] = t.get('begin', transcripts.time_missing)
-					t['end'] = t.get('end', transcripts.time_missing)
-					t['channel'] = t.get('channel', transcripts.channel_missing)
-
-				examples_filtered.append(example)
-				transcript.extend(example)
-				examples_lens.append(len(example))
-		
-		self.speaker_names = transcripts.collect_speaker_names(transcript, speaker_names = speaker_names or [], num_speakers = max_num_channels, set_speaker = True)
-
-		_print('Dataset construction time: ', time.time() - tic); tic = time.time()
-		
-		self.bucket = torch.ShortTensor([e[0]['bucket'] for e in examples_filtered]) 
-		self.audio_path = utils.TensorBackedStringArray([e[0]['audio_path'] for e in examples_filtered], encoding = string_array_encoding)
-		self.ref = utils.TensorBackedStringArray([t['ref'] for t in transcript], encoding = string_array_encoding)
-		self.begin = torch.DoubleTensor([t['begin'] for t in transcript])
-		self.end = torch.DoubleTensor([t['end'] for t in transcript])
-		self.channel = torch.CharTensor([t['channel'] for t in transcript])
-		self.speaker = torch.LongTensor([t['speaker'] for t in transcript])
-		self.cumlen = torch.ShortTensor(examples_lens).cumsum(dim = 0, dtype = torch.int64)
+			if exclude is not None:
+				allowed_audio_names = set(transcripts.audio_name(t) for t in transcript if transcripts.audio_name(t) not in exclude)
+			else:
+				allowed_audio_names = None
+
+			transcript = transcripts.prune(transcript,
+			                               allowed_audio_names = allowed_audio_names,
+			                               duration = (min_duration if min_duration is not None else 0.0,
+			                                           max_duration if max_duration is not None else 24.0 * 3600,), #24h
+			                               max_audio_file_size = max_audio_file_size)
+			transcript = list(transcript)
+			for t in transcript:
+				t['example_id'] = AudioTextDataset.get_example_id(t)
+
+			if len(transcript) == 0:
+				continue
+
+			bucket = bucket_fn(transcript)
+			for t in transcript:
+				t['bucket'] = bucket
+			buckets.append(bucket)
+			grouped_segments.extend(transcript)
+			transcripts_len.append(len(transcript))
+
+		_print('Dataset construction time: ', time.time() - tic)
+		tic = time.time()
+
+		self.bucket = torch.tensor(buckets, dtype = torch.short, device = 'cpu')
+		self.audio_path = utils.TensorBackedStringArray([t['audio_path'] for t in grouped_segments], encoding = string_array_encoding)
+		self.ref = utils.TensorBackedStringArray([t['ref'] for t in grouped_segments], encoding = string_array_encoding)
+		self.begin = torch.tensor([t['begin'] for t in grouped_segments], dtype = torch.float64, device = 'cpu')
```

Comment:
are these still needed? we discussed that this information could remain in `meta` only